### ISSUE TITLE
Warlock rank 3 DoT spells has reduced scaling on spell power. CoA r3 does too much damage baseline.

### ISSUE THREAD

**Issue by Tavta** [slowtorta/turtlewow-bug-tracker#2445]
**Please, provide a clear description what the bug is:**
Curse of Agony r3, Corruption r3, Immolate r3 do not have the spell power coefficients they should have at that rank.
CoA r3 should have 100%, but has 43,4%. Corr r3 should have 100%, but has 78,2%. Immolate should have combined 70%, but has 61%.

CoA has too high base damage values.
CoA r3 should do 324 dmg over 24 sec. I observe values between 379-380.

**Steps to reproduce the behavior with as much detail as possible:**

1. Get 23 spell power. (I used Charged Leggins, Magnificent Necklace and Staff of Soran'Ruk)
2. Compare spells with and without said items. Let the dots tick out fully.
3. Test on different mobs (I used crocolisks near crypt and black ooze slimes outside of crypt in Wetlands)

**IDs of affected creatures, items, quests or spells with a link to the relevant page:**
https://www.wowhead.com/classic/spell=6217/curse-of-agony#see-also-ability CoA spell coefficient from vanilla classic
https://www.
wowhead.com/classic/spell=6223/corruption Corruption spell coefficient from vanilla classic
https://www.wowhead.com/classic/spell=1094/immolate Immolate spell coefficient from vanilla classic
![WarlockDmgCalcsDump](https://user-images.githubusercontent.com/117946241/201202816-67dc9d1c-1fd7-44ad-9261-653add43da4c.png)

NPC: https://database.turtle-wow.org/?npc=
Item: https://database.turtle-wow.org/?item=
Quest: https://database.turtle-wow.org/?quest=

Spell: https://database.turtle-wow.org/?spell=
https://database.turtle-wow.org/?spell=6223
https://database.turtle-wow.org/?spell=6217
https://database.turtle-wow.org/?spell=1094

Object: https://database.turtle-wow.org/?object=

**Expected behavior. Describe how it should work:**
Spells should scale up to maximum capacity for spell power as they're learnable at lvl 20 or later (they are no longer penalized by the below lvl 20 clause).

**Comment by sieleyah** [comment]
The wiki page on downranking:

_"A penalty for downranking to spells below level 20 has been present since early in the game's development. Shortly before the release of the 2.0 content patch, an additional downranking modifier was added in order to combat significant downranking of [healing](https://vanilla-wow-archive.fandom.com/wiki/Heal) spells in [raiding](https://vanilla-wow-archive.fandom.com/wiki/Raid). With the original system, as long as the spell was over level 20, there was no penalty. This led to situations where certain classes were using spells down to into the lower 20s and compensating for the lower base healing amount by stacking +heal stats, resulting in an extreme increases in healing per [mana](https://vanilla-wow-archive.fandom.com/wiki/Mana) beyond what Blizzard's encounter design had intended.

The new formula scales down bonus healing and damage for spells which could have been learned more than 11 levels ago, by a simple multiplier:

Calculation:
sLvl = Level that the rank of this spell can first be learned
cLvl = Caster's Current Level
Downranking Multiplier = Min(1, (sLvL + 11) / cLvl)"_

This came out before BC and I believe this is what turtle uses. It checks for the gap between rank lvl and player lvl. 

The rest of the report seems valid. COA is wrongly staged and the increased values of total damage you experience stems from the first 4 ticks doing the same damage as the #4-8 ticks. Instead of 4 low, 4 average, 4 high damage ticks, we have 8 average and 4 high.

**Comment by Tavta** [comment]
R3 of Corr, Immo and CoA are all above the level 20 penalty. If you click on those links I’ve presented you would be able to see that Immolation is learned at lvl 20, Corruption at lvl 24, CoA lvl 28. Again, I’m presenting the rank3 version of these spells.

**Comment by sieleyah** [comment]
Please read my post again. The level 20 penalty is not relevant. If your character is more than 11 levels above the spellranks, they will receive penalty.
For rank 3 COA which is learned at lvl 28, this penalty starts applying when your character hits lvl 39.

**Comment by Tavta** [comment]
>

> Please read my post again. The level 20 penalty is not relevant. If your character is more than 11 levels above the spellranks, they will receive penalty. For rank 3 COA which is learned at lvl 28, this penalty starts applying when your character hits lvl 39.

You're talking about the prepatch before tbc? The patch we are not playing on? How would that matter for twow?
https://wowwiki-archive.fandom.com/wiki/Downranking

**Comment by sieleyah** [comment]
We are playing on a custom patch with many BC features implemented. Nonetheless it appears that it was wrongly implemented if so.
I just tested the ranks myself on a lvl 60 warlock and it appears I am not getting any penalty on r3-r4 of these spells at all. If you experienced those values at lvl 28 that could mean the scaling checks for character level instead of the gap between the rank and the character. So lowbies are getting penalized while max levels are not. Poor lowbies. At this point they should just remove the system and keep the 20 lvl penalty.
So your report is valid but it requires a low level character to reproduce the issue.

**Comment by sieleyah** [comment]
Ok that makes more sense. So there is nothing wrong with the coefficient values nor did they implement prepatch scaling (at least at lvl 60 for sure) but the check point of the dots are per-tick instead of per-ability. I'm pretty sure the spell should calculate the full damage first the bonus first and then divide accordingly, rather than adding damage per-tick (which causes a loss of small spell power values as the per-tick coefficient is too small to provide a whole number unless your SP is a multiple of 6 or 12)

**Comment by Tavta** [comment]
> Ok that makes more sense. So there is nothing wrong with the coefficient values nor did they implement prepatch scaling (at least at lvl 60 for sure) but the check point of the dots are per-tick instead of per-ability. I'm pretty sure the spell should calculate the full damage with the bonus first and then divide accordingly, rather than adding damage per-tick (which causes a loss of small spell power values as the per-tick coefficient is too small to provide a whole number unless your SP is a multiple of 6 or 12) EDIT: this could also just be an issue with rounding down on values that should be rounded up
> 
> I suggest editing your post to address this issue because they won't be able to reproduce a problem with the spell coefficient itself.

You're right, I should edit my original post to better reflect the current way I'm viewing these scaling issues.
